server/src/main/java/org/elasticsearch/cluster/project/ProjectStateRegistry.java

```diff
@@ -304,14 +316,22 @@ private Builder(ProjectStateRegistry original) {
             this.projectsMarkedForDeletionGeneration = original.projectsMarkedForDeletionGeneration;
         }
 
-        public Builder putProjectSettings(ProjectId projectId, Settings settings) {
+        private void modifyEntry(ProjectId projectId, Function<Entry, Entry> modifier) {
```

Comment:
Nit: I think similar existing methods tend to use `update` instead of `modify`, i.e. `updateEntry`.